### PR TITLE
Created the comment section

### PR DESCRIPTION
**Pull request by zirving**
Completed Step 4: create the comment section 

- Comment.java : helper class for comments - contains a string for a username and a comment

- index.html: Added form section where users can submit their username and a comment - calls doPost() of DataServlet. Added an existing comments section which displays the existing comments by calling displayComments()

- script.js: removed old getDatafromServlet. Added displayComments() which outputs a message if there are no comments, and calls createCommentElement for each existing comment. createCommentElement generates a header (username), paragraph (comment), and dividing line. 

- DataServlet: when a comment is submitted, adds it to an ArrayList of comments (comment history). returns the comment history when a doGet() is called. 

-style.css: added formatting for comment section

image

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,29 +19,54 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> data = new ArrayList<String>();
-    data.add("Value 1");
-    data.add("Value 2");
-    data.add("Value 3");
-
-    String json = convertToJsonUsingGson(data);
+    if(commentHistory.isEmpty()){
+        response.setContentType("text/html");
+        response.getWriter().println("-1");
```
Comment (ganewman42) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: -1 isn't a super helpful error message to display to the client. Something like "No Comments Found" might be more helpful.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -19,29 +19,54 @@
...
     Gson gson = new Gson();
     String json = gson.toJson(data);
     return json;
   }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (ganewman42) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:63]: This is a nice little helper, good idea.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -58,11 +58,42 @@ function loadMusicVideo(){
...
-     dataElement.innerHTML = '';
-     dataElement.appendChild(document.createTextNode(" " + data[0]));
-     dataElement.appendChild(document.createTextNode(" " + data[1]));
-     dataElement.appendChild(document.createTextNode(" " + data[2]));
+  fetch('/data').then(response => response.json()).then((data) => { 
+    const dataElement = document.getElementById("servlet-msg-container");
+    dataElement.innerHTML = '';
+    dataElement.appendChild(document.createTextNode(" " + data[0]));
+    dataElement.appendChild(document.createTextNode(" " + data[1]));
+    dataElement.appendChild(document.createTextNode(" " + data[2]));
```
Comment (ganewman42) [inline portfolio/src/main/webapp/script.js]: You mentioned removing this in PR #5, did you forget to take it out?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -19,29 +19,54 @@
...
+    }
+    String json = convertToJsonUsingGson(commentHistory);
     
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  private String convertToJsonUsingGson(ArrayList<String> data) {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String username = getParameter(request,"username-input","Anonymous");
```
Comment (cthamrun) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: spaces after commas (i.e. request, "username-input", "Anonymous")

**Review by ganewman42 — COMMENTED** [review]
Some minor revisions on error handling and what looks like a section that should have been removed.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -19,29 +19,54 @@
...
+      String comment = getParameter(request, "comment-input","");
+      commentHistory.add(new Comment(username,comment));
```
Comment (cthamrun) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: ditto as above

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -19,29 +19,54 @@
...
+    String json = convertToJsonUsingGson(commentHistory);
     
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
-  private String convertToJsonUsingGson(ArrayList<String> data) {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      String username = getParameter(request,"username-input","Anonymous");
+      String comment = getParameter(request, "comment-input","");
```
Comment (cthamrun) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: ditto as above

**Review by cthamrun — APPROVED** [review]
Just a few minor comments but everything else looks good.

**Pull request merged**